internal/gengapic/generator.go

```diff
@@ -99,17 +101,22 @@ func (g *generator) init(req *plugin.CodeGeneratorRequest) error {
 	files := req.GetProtoFile()
 
 	if opts.serviceConfigPath != "" {
-		f, err := os.Open(opts.serviceConfigPath)
+		y, err := ioutil.ReadFile(opts.serviceConfigPath)
 		if err != nil {
-			return errors.E(nil, "error opening service config: %v", err)
+			return errors.E(nil, "error reading service config: %v", err)
 		}
-		defer f.Close()
 
-		g.serviceConfig = &serviceconfig.Service{}
-		err = yaml.NewDecoder(f).Decode(g.serviceConfig)
+		j, err := yaml.YAMLToJSON(y)
 		if err != nil {
-			return errors.E(nil, "error decoding service config: %v", err)
+			return errors.E(nil, "error converting YAML to JSON: %v", err)
+		}
+
+		cfg := &serviceconfig.Service{}
+		unm := protojson.UnmarshalOptions{DiscardUnknown: true}
+		if err := unm.Unmarshal(j, cfg); err != nil {
```

Comment:
nit: this and the above line could be combined.

Reply:
Done.